core/src/scene/drawRule.h

```diff
@@ -38,15 +53,39 @@ struct DrawRule {
         _value = param.value.get<T>();
         return true;
     }
+
     bool contains(StyleParamKey _key) const {
         return findParameter(_key) != false;
     }
 
-    bool operator<(const DrawRule& _rhs) const;
-    int compare(const DrawRule& _rhs) const { return name.compare(_rhs.name); }
-
 private:
-    void logGetError(StyleParamKey _expectedKey, const StyleParam& _param);
+    void logGetError(StyleParamKey _expectedKey, const StyleParam& _param) const;
+
+};
+
+struct Styling {
+
+    /* Determine and apply DrawRules for a @_feature and add
+     * the result to @_tile
+     */
+    void apply(const Feature& _feature, const Scene& _scene,
+               const SceneLayer& _sceneLayer,
+               StyleContext& _ctx, Tile& _tile);
+
+    // internal
+    bool match(const Feature& _feature, const SceneLayer& _layer, StyleContext& _ctx);
+
+    // internal
+    void mergeRules(const std::vector<StaticDrawRule>& rules);
```

Comment:
`apply`, `match` and `mergeRules`: potential candidate for inline?

Don't think it will lead a lot of performance improvement, but something to think about based on their usage.